src/entities/DFGVariable.jl

```diff
@@ -10,7 +10,7 @@ mutable struct VariableNodeData
   eliminated::Bool
   BayesNetVertID::Symbol #  Union{Nothing, }
   separator::Array{Symbol,1}
-  softtype
+  softtype  # Perhaps this should move up to DFGVariable level
```

Comment:
I would think only the `softtypename::Symbol`
but I added a softtype(v::DFGVariable) to maintain compatibility.

Reply:
see #149

Reply:
> should DFG not export an abstract type with IIF InferenceType inheriting from it?

DFG already exports `InferenceType` and `InferenceVariable`.  `InferenceType` is already an abstract type, so dont think we need to add another layer. https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/149#issuecomment-540642450